pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -265,19 +284,193 @@ func TestReconcile(t *testing.T) {
+			expected: true,
+		},
+	}
+
+	for _, tc := range testsCases {
+		if got := hasMachineSetOwner(*tc.machine); got != tc.expected {
+			t.Errorf("Test case: Machine %s. Expected: %t, got: %t", tc.machine.Name, tc.expected, got)
+		}
+	}
+
+}
+
+func TestLastTransitionTime(t *testing.T) {
+	node := node("test", true)
+	testsCases := []struct {
+		node      *v1.Node
+		condition corev1.NodeConditionType
+		expected  metav1.Time
+	}{
+		{
+			node:      node,
+			condition: corev1.NodeReady,
+			expected:  knownDate,
+		},
+		{
+			node:      node,
+			condition: corev1.NodeOutOfDisk,
+			expected:  metav1.Time{Time: metav1.Now().Add(-2 * remediationWaitTime)},
+		},
+	}
+
+	for _, tc := range testsCases {
+		got := lastTransitionTime(tc.node, tc.condition)
+		if got.Sub(tc.expected.Time) > time.Second {
+			t.Errorf("Expected %s, got %s", tc.expected.String(), got.String())
+		}
+	}
+}
+
+func TestUnhealthyForTooLong(t *testing.T) {
+	nodeUnhealthyForTooLong := node("nodeUnhealthyForTooLong", false)
+	nodeRecentlyUnhealthy := node("nodeRecentlyUnhealthy", false)
+	nodeRecentlyUnhealthy.Status.Conditions[0].LastTransitionTime = metav1.Time{Time: time.Now()}
+	testsCases := []struct {
+		node     *corev1.Node
+		expected bool
+	}{
+		{
+			node:     nodeUnhealthyForTooLong,
+			expected: true,
+		},
+		{
+			node:     nodeRecentlyUnhealthy,
+			expected: false,
+		},
+	}
+	for _, tc := range testsCases {
+		if got := unhealthyForTooLong(tc.node); got != tc.expected {
+			t.Errorf("Test case: %s. Expected: %t, got: %t", tc.node.Name, tc.expected, got)
+		}
+	}
+}
+
+func TestRemediate(t *testing.T) {
+	nodeHealthy := node("nodeHealthy", true)
+	nodeHealthy.Annotations = map[string]string{
+		"machine": fmt.Sprintf("%s/%s", namespace, "machineWithNodehealthy"),
+	}
+
+	nodeRecentlyUnhealthy := node("nodeRecentlyUnhealthy", false)
+	nodeRecentlyUnhealthy.Status.Conditions[0].LastTransitionTime = metav1.Time{Time: time.Now()}
+	nodeRecentlyUnhealthy.Annotations = map[string]string{
+		"machine": fmt.Sprintf("%s/%s", namespace, "machineWithNodeRecentlyUnhealthy"),
+	}
+
+	machineWithNodeRecentlyUnhealthy := machine("machineWithNodeRecentlyUnhealthy")
+	machineWithNodeRecentlyUnhealthy.Status = capiv1alpha1.MachineStatus{
+		NodeRef: &corev1.ObjectReference{
+			Namespace: "",
+			Name:      "nodeRecentlyUnhealthy",
+		},
+	}
+
+	machineWithNodeHealthy := machine("machineWithNodehealthy")
+	machineWithNodeHealthy.Status = capiv1alpha1.MachineStatus{
+		NodeRef: &corev1.ObjectReference{
+			Namespace: "",
+			Name:      "nodeHealthy",
+		},
+	}
+
+	machineWithNoOwnerController := machine("machineWithNoOwnerController")
+	machineWithNoOwnerController.OwnerReferences = nil
+
+	machineWithNoNodeRef := machine("machineWithNoNodeRef")
+
+	testsCases := []struct {
+		machine  *capiv1alpha1.Machine
+		expected expectedReconcile
+	}{
+		{
+			machine: machineWithNodeHealthy,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
```

Comment:
I would advocate using the "zero" value here, and for the other cases that default to `false`. Omitting this entry will default to `false`.

Reply:
I'd rather keep it explicitly for the test cases, it makes it easier to visualise and understand the expectation when looking at it